### PR TITLE
Build: Use Spark 3.2 for cross version verification

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

Follow up #1876

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Review by SteNicholas — APPROVED** [review]
LGTM. cc @yaooqinn

**Comment by pan3793** [comment]
Thanks, merging to master

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -23,21 +23,21 @@ jobs:
             spark-hadoop: '2.7'
           - java: 8
             spark: '3.1'
+            spark-hadoop: '3.2'
+          - java: 8
+            spark: '3.2'
```
Comment (yaooqinn) [inline .github/workflows/master.yml:28]: `spark-hadoop: '2.7'` means breaking hadoop3.2?

Comment (pan3793) [reply .github/workflows/master.yml:28]: I don't get your ponit.

Previously, we use `kyuubi-spark-sql-engine.jar` build against Spark 3.1 do corss test on spark-3.0 & hadoop-2.7 and spark 3.2 & hadoop 3.2.

Now, we use `kyuubi-spark-sql-engine.jar` build against Spark 3.2 do corss test on spark-3.0 & hadoop-2.7 and spark 3.1 & hadoop 3.2.

Comment (yaooqinn) [reply .github/workflows/master.yml:28]: I mean cross check shall use default profiles

Comment (pan3793) [reply .github/workflows/master.yml:28]: The default hadoop version of Spark 3.1 is 3.2. https://mvnrepository.com/artifact/org.apache.spark/spark-core_2.12/3.1.3

Comment (yaooqinn) [reply .github/workflows/master.yml:28]: shall we use 3.2 instead of 2.7

**Pull request closed without merging**